pkg/server/test/list_objects.go

```diff
@@ -308,132 +334,88 @@ func TestListObjectsRespectsMaxResults(t *testing.T, ds storage.OpenFGADatastore
-					"viewer": typesystem.This(),
-				},
-				Metadata: &openfgav1.Metadata{
-					Relations: map[string]*openfgav1.RelationMetadata{
-						"viewer": {
-							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
-								typesystem.DirectRelationReference("user", ""),
-							},
-						},
-					},
-				},
-			},
-		},
+		TypeDefinitions: parser.MustParse(`
+	type user
+
+	type document
+	  relations
+	    define viewer: [user] as self
+	`),
 	}
-	err := ds.WriteAuthorizationModel(ctx, store, model)
+	err := ds.WriteAuthorizationModel(context.Background(), storeID, model)
 	require.NoError(b, err)
 
-	n := 0
+	numberObjectsAccesible := 0
 	for i := 0; i < 100; i++ {
 		var tuples []*openfgav1.TupleKey
 
 		for j := 0; j < ds.MaxTuplesPerWrite(); j++ {
-			obj := fmt.Sprintf("document:%s", strconv.Itoa(n))
-			user := fmt.Sprintf("user:%s", strconv.Itoa(n))
+			obj := fmt.Sprintf("document:%s", strconv.Itoa(numberObjectsAccesible))
 
-			tuples = append(tuples, tuple.NewTupleKey(obj, "viewer", user))
+			tuples = append(tuples, tuple.NewTupleKey(obj, "viewer", "user:maria"))
 
-			n += 1
+			numberObjectsAccesible += 1
 		}
 
-		err = ds.Write(ctx, store, nil, tuples)
+		err := ds.Write(context.Background(), storeID, nil, tuples)
 		require.NoError(b, err)
 	}
 
-	listObjectsQuery := commands.NewListObjectsQuery(ds)
-
-	var r *commands.ListObjectsResponse
-
-	ctx = typesystem.ContextWithTypesystem(ctx, typesystem.New(model))
-
-	b.ResetTimer()
-	for i := 0; i < b.N; i++ {
-		r, _ = listObjectsQuery.Execute(ctx, &openfgav1.ListObjectsRequest{
-			StoreId:              store,
-			AuthorizationModelId: model.Id,
-			Type:                 "document",
-			Relation:             "viewer",
-			User:                 "user:999",
-		})
-	}
-
-	listObjectsResponse = r
+	return model, modelID, numberObjectsAccesible
 }
 
-func BenchmarkListObjectsWithConcurrentChecks(b *testing.B, ds storage.OpenFGADatastore) {
+func BenchmarkListObjects(b *testing.B, ds storage.OpenFGADatastore) {
 	ctx := context.Background()
 	store := ulid.Make().String()
 
-	typedefs := parser.MustParse(`
-	type user
-
-	type document
-	  relations
-	    define allowed: [user] as self
-	    define viewer: [user] as self and allowed
-	`)
+	model, modelID, numberObjectsAccessible := setupListObjectsBenchmark(b, ds, store)
+	ctx = typesystem.ContextWithTypesystem(ctx, typesystem.New(model))
 
-	model := &openfgav1.AuthorizationModel{
-		Id:              ulid.Make().String(),
-		SchemaVersion:   typesystem.SchemaVersion1_1,
-		TypeDefinitions: typedefs,
+	req := &openfgav1.ListObjectsRequest{
+		StoreId:              store,
+		AuthorizationModelId: modelID,
+		Type:                 "document",
+		Relation:             "viewer",
+		User:                 "user:maria",
 	}
-	err := ds.WriteAuthorizationModel(ctx, store, model)
-	require.NoError(b, err)
 
-	n := 0
-	for i := 0; i < 100; i++ {
-		var tuples []*openfgav1.TupleKey
+	b.ResetTimer()
```

Comment:
Shouldn't `b.ResetTimer` also be nested within the sub-benchmarks before the bechnmark loop. e.g.

```go
b.Run("oneResult", func(b *testing.T) {
    ...
    b.ResetTimer()
    for i := o; i < b.N; i++ {
       // benchmark under test
    }
})
```

Otherwise we'll be timer benchmark setup time for the sub-benchmarks runs between `b.Run("oneResult", ...)` and `b.Run("allResults", ...)`.